projects/igniteui-angular/src/lib/grids/cell.component.ts

```diff
@@ -889,6 +889,14 @@ export class IgxGridCellComponent implements OnInit, OnChanges, OnDestroy, CellT
                 this.highlight.lastSearchInfo.caseSensitive = this.grid.lastSearchInfo.caseSensitive;
                 this.highlight.lastSearchInfo.exactMatch = this.grid.lastSearchInfo.exactMatch;
             }
+
+            // add check in case value changes externally and form control needs to update.
+            if (!this.editMode && this.formControl && this.formControl.value != changes.value.currentValue) {
```

Comment:
While this will certainly fix the specific scenario described in the issue, it relies on the cell being in view to fix the state, which might not always be the case. If the cell is out of view when the data changes (out of scroll or another page, etc) this won't happen and leave the validation state of the grid out of sync until the cell comes in view.

So it stands to reason the cell component shouldn't be handling the state update, just displaying it. The state update should be driven by the data, so when that changes any existing row states affected should be updated with the new data w/ any possible transactions active on top

Reply:
@damyanpetev  That will be rather hard to do without re-running validation on all formControls that have matching record id in the new data since we don’t really know if the data change was due to you wanting to set a fully new data to the grid, do some remote operation (like remote paging, filtering etc.)  or maybe do some live data update in existing records.

Would we be ok with that?

Reply:
Yeah I don't think we distinguish between those uses. Setting data generally not something that happens in a rapid succession so the hit shouldn't be that significant. I see no other option but to check the new data for cached validation state keys and update the state if needed. And there is a soft check to avoid running validation if the value is the same.. or at least used to be one, cuz I think it just got removed for some reason.

BTW, for the issue's use case if the goal is to 'refresh' the grid to initial state, a call to the validation service to clear existing state could probably be part of that. Perhaps something we should offer to be used.